### PR TITLE
Project 40

### PR DESCRIPTION
**Pull request by Abdelrhman0z**

### CODE REVIEW

File: 40- 3D Boxes Background/script.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    for (let j = 0; j < 4; j++) {
+      const box = document.createElement('div')
+      box.classList.add('box')
+      box.style.backgroundPosition = `${-j * 125}px ${-i * 125}px`
+      boxes.appendChild(box)
+    }
+  }
+}
+
+createBoxes()
```
Comment (a-soliman) [inline 40- 3D Boxes Background/script.js]: Did you come across a concept called "IIFE" (Immediately Invoked Function Expression)? not a biggie but could be used here, take a few minutes to read about it.

**Review by a-soliman — APPROVED** [review]
Looks good.

File: 40- 3D Boxes Background/script.js (hunk 1 above)
Comment (Abdelrhman0z) [reply 40- 3D Boxes Background/script.js]: so i read about it and i think it's cool , but however there is an issue , when i try and apply it on my code the function dose not work until i remove this two lines
`const boxes = document.getElementById('boxes') 
const btn = document.getElementById('btn')`

so is that normal or what ?

Comment (a-soliman) [reply 40- 3D Boxes Background/script.js]: nope, that isn't normal, I think something must be wrong with your syntax 
try the following
```
const boxes = document.getElementById('boxes')
const btn = document.getElementById('btn')

(function createBoxes() {
for (let i = 0; i < 4; i++) {
    for (let j = 0; j < 4; j++) {
      const box = document.createElement('div')
      box.classList.add('box')
      box.style.backgroundPosition = `${-j * 125}px ${-i * 125}px`
      boxes.appendChild(box)
    }
  }
})()
```

Comment (Abdelrhman0z) [reply 40- 3D Boxes Background/script.js]: nope , still the same issue , it gives me this error  **Uncaught TypeError: document.getElementById(...) is not a function**
i tried to change it to querySelector but still the same problem 😵🤔

Comment (a-soliman) [reply 40- 3D Boxes Background/script.js]: cool no worries you can proceed with what you have, tho a BOUNUS point if you can identify why it does what it does. 
Hint: its one of these few cases where `;` actually has a meaning in JS

**Review by a-soliman — APPROVED** [review]

**Pull request merged**